File: api/messages.js
import { MongoClient, ObjectId } from "mongodb";

const client = new MongoClient(process.env.MONGO_URL);
const dbName = process.env.DB_NAME || "portfolio_db";

export default async function handler(req, res) {
  if (req.method === "GET") {
    try {
      await client.connect();
      const db = client.db(dbName);
      const messages = db.collection("messages");

      const docs = await messages.find().toArray();
      res.status(200).json(docs);
    } catch (err) {
      res.status(500).json({ error: err.message });
    }
  } else {
    res.status(405).json({ error: "Method not allowed" });
  }
}

File: api/messages.py
from pymongo import MongoClient
import os, json
from bson import ObjectId

MONGO_URL = os.environ["MONGO_URL"]
DB_NAME = os.environ.get("DB_NAME", "portfolio_db")

client = MongoClient(MONGO_URL)
db = client[DB_NAME]
messages = db["messages"]

def handler(request, response):
    if request.method == "GET":
        try:
            docs = list(messages.find())
            for d in docs:
                d["_id"] = str(d["_id"])  # Convert ObjectId to string
            return response.json(docs)
        except Exception as e:
            return response.json({"error": str(e)}, status=500)
    return response.json({"error": "Method not allowed"}, status=405)
